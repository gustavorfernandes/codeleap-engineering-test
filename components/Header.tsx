function Header() {
  return (
    <div className="w-screen max-w-[800px] bg-black h-16 sm:h-20 flex items-center justify-center">
      <div className="w-11/12 flex items-center">
        <h2 className="font-roboto text-sm sm:text-xl md:text-[22px] font-bold text-white">
          CodeLeap Network
        </h2>
      </div>
    </div>
  )
}

export default Header
