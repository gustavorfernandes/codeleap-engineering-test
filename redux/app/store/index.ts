import { configureStore } from "@reduxjs/toolkit"
import usernameReducer from "../slices/usernameSlice"

export const store = configureStore({
  reducer: {
    username: usernameReducer
  },
});

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = ReturnType<typeof store.dispatch>
