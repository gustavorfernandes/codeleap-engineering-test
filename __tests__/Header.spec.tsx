import { makeStore } from '../redux/app/store/index'
import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import Header from "../components/Header"

describe("Header component", () => {
  it("The component must render correctly", () => {
    const store = makeStore()
    const { container } = render(
      <Provider store={store} >
        <Header />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})
