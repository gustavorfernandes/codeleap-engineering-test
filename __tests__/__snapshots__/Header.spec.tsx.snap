// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component The component must render correctly 1`] = `
<div>
  <div
    class="w-screen max-w-[800px] bg-black h-16 sm:h-20 flex items-center justify-center"
  >
    <div
      class="w-11/12 flex items-center"
    >
      <h2
        class="font-roboto text-sm sm:text-xl md:text-[22px] font-bold text-white"
      >
        CodeLeap Network
      </h2>
    </div>
  </div>
</div>
`;
